packages/react/src/components/cookieConsentCore/translations.js

```diff
@@ -0,0 +1,118 @@
+
+  const translationNotFound = !translations[key];
+  const translationIsAnArray = Array.isArray(translations[key]);
+  const translationIsObjectAndHasNoKeys =
+    typeof translations[key] === 'object' && !translationIsAnArray && Object.keys(translations[key]).length === 0;
+
+  // Tries to find translation in
+  //   given language,
+  //   fallback language or
+  //   first available language,
+  // in that order, returns translation key if not found
+
+  if (translationNotFound || translationIsObjectAndHasNoKeys) {
+    // If translation is missing, return key
+
+    translation = key;
+    // eslint-disable-next-line no-console
+    console.error(`Cookie consent: Missing translation key: ${key}, falling back to key as translation`);
+  } else if (typeof translations[key] === 'string' || typeof translations[key] === 'number') {
+    // Same translation for all languages.
+    translation = translations[key];
+  } else if (typeof translations[key] === 'object' && !translationIsAnArray) {
+    // Different translations for different languages.
+    if (translations[key][lang]) {
+      // Translation was found in given language, use it
+      translation = translations[key][lang];
+    } else if (translations[key][fallbackLang]) {
+      // Translation was not found in given language, but it was found in fallback language, use it
+      translation = translations[key][fallbackLang];
+      // eslint-disable-next-line no-console
+      console.error(`Cookie consent: Missing translation: ${key}:${lang}, using fallback language: ${fallbackLang}`);
+    } else {
+      // Translation was not found in given language or fallback language, use first available translation
+      const firstLang = Object.keys(translations[key])[0];
+      translation = translations[key][firstLang];
+      // eslint-disable-next-line no-console
+      console.error(
+        `Cookie consent: Missing primary and fallback translation: ${key}:${lang}/${fallbackLang}, using first known language: ${firstLang}`,
+      );
+    }
+  } else {
+    // Translation is not a string, number or object
+    throw new Error(`Cookie consent: Invalid translation: ${key}, should be string, number or object`);
+  }
+
+  if (translation) {
+    // Replace dollar strings in translation with corresponding data from parameters
+    return translation.replace(/\$\{.+?\}/g, (match) => {
+      const stripDollarAndParenthesis = match.replace(/(^\$\{|\}$)/g, '');
+      const parameter = index(parameters, stripDollarAndParenthesis);
+
+      // Parameters may be either string or an language object
+      if (typeof parameter === 'object') {
+        if (parameter[lang]) {
+          return parameter[lang];
+        }
+
+        // Use fallback language if translation is missing
+        if (parameter[fallbackLang]) {
+          return parameter[fallbackLang];
+        }
+
+        // Use first available language if translation is missing
+        const firstLang = Object.keys(parameter)[0];
```

Comment:
The key is not meant to be read by users. It's purely for internal use. This translation thing is trying to be as fault tolerant to the end user as possible.

Reply:
The key is returned in other cases too

```
 if (translationNotFound || translationIsObjectAndHasNoKeys) {
    // If translation is missing, return key
    translation = key;
```

So I thought it would be better to return the key rather than text in other language. 

If this is common way in hel.fi, then it is fine.